subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2023,15 +2014,88 @@ static int lwm2m_delete_handler(struct lwm2m_engine_obj *obj,
 				     context->path->obj_inst_id);
 }
 
+static int get_coap_packet_len(struct net_pkt *pkt)
+{
+	struct net_buf *frag;
+	u16_t offset;
+	u16_t len;
+
+	frag = net_frag_read_be16(pkt->frags,
+				  net_pkt_ip_hdr_len(pkt) +
+				  net_pkt_ipv6_ext_len(pkt) +
+				  4, &offset, &len);
+	if (!frag && offset == 0xffff) {
+		return -1;
+	}
+
+	return len - NET_UDPH_LEN;
+}
+
+#define COAP_MARKER 0xFF
+
+/* TODO: Remove in favor of using net_buf pointer to payload */
+u8_t *coap_packet_get_payload_ptr(struct coap_packet *cpkt, u16_t *len,
```

Comment:
Is "coap_packet_get_payload()" api not helpful for your requirement?

Reply:
This is a temporary function to preserve functionality for the LwM2M content formatters until I re-work them to use multi fragments.  They go away once that's complete.